provider/openstack/credentials.go

```diff
@@ -70,7 +86,20 @@ func (OpenstackCredentials) CredentialSchemas() map[cloud.AuthType]cloud.Credent
 					Hidden:      true,
 				},
 			}, {
-				CredAttrTenantName, cloud.CredentialAttr{Description: "The OpenStack tenant name."},
+				CredAttrTenantName, cloud.CredentialAttr{
+					Description: "The OpenStack tenant name.",
+					Optional:    true,
+				},
+			}, {
+				CredAttrTenantID, cloud.CredentialAttr{
+					Description: "The Openstack tenant ID",
+					Optional:    true,
+				},
+			}, {
+				CredAttrVersion, cloud.CredentialAttr{
+					Description: "The Openstack keystone version",
```

Comment:
s/keystone/identity/

This is typically referred to as the identity version.  The keystone service is configured to use a specific identity api version.  The env var is referred to as OS_IDENTITY_API_VERSION